TCC.Lib/CommandHelper.cs

```diff
@@ -112,87 +113,116 @@ public static bool IsIgnorable(string line)
-                        {
-                            errorWaitHandle.Set();
-                        }
-                        else
-                        {
-                            if (IsIgnorable(e.Data))
-                            {
-                                return;
-                            }
-                            if (IsInfo(e.Data))
-                            {
-                                result.Infos.Add(e.Data);
-                            }
-                            else
-                            {
-                                error.AppendLine(e.Data);
-                            }
-                        }
-                    };
+                        standardInfos.Add(e.Data);
+                    }
+                    else
+                    {
+                        standardError.Add(e.Data);
+                    }
+                }
+            };
 
-                    process.Start();
+            var tcs = new TaskCompletionSource<CommandResult>();
 
-                    cancellationToken.Register(() =>
-                    {
-                        KillProcessAndChildren(process.Id);
-                    });
+            async void OnProcessExit(object sender, EventArgs args)
+            {
+                var stdout = await standardOutputResults.Task;
+                var stderr = await standardErrorResults.Task;
 
-                    process.BeginOutputReadLine();
-                    process.BeginErrorReadLine();
+                var result = new CommandResult
+                {
+                    Command = command,
+                    ExitCode = process.ExitCode,
+                    IsSuccess = standardError.Count == 0,
+                    Output = string.Join(Environment.NewLine, stdout),
+                    Errors = string.Join(Environment.NewLine, stderr),
+                    Infos = standardInfos,
+                };
+
+                tcs.TrySetResult(result);
+                process.Exited -= OnProcessExit;
+            }
 
-                    int timeout = (int)timeoutLimit.TotalMilliseconds;
+            process.Exited += OnProcessExit;
 
-                    if (process.WaitForExit(timeout) &&
-                        outputWaitHandle.WaitOne(timeout) &&
-                        errorWaitHandle.WaitOne(timeout))
+            using (cancellationToken.Register(() =>
+            {
+                tcs.TrySetCanceled();
+                process.Exited -= OnProcessExit;
+                try
+                {
+                    KillProcessAndChildren(process.Id);
+
+                    if (!process.HasExited)
+                    {
+                        process.Kill();
+                    }
+                }
+                catch (InvalidOperationException)
+                {
+                }
+            }))
+            {
+                try
+                {
+                    if (process.Start() == false)
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the unnecessary Boolean literal(s). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS1125)